bugwarrior/services/bts.py

```diff
@@ -0,0 +1,175 @@
+    URL = 'btsurl'
+    NUMBER = 'btsnumber'
+    PACKAGE = 'btspackage'
+    SOURCE = 'btssource'
+    FORWARDED = 'btsforwarded'
+
+    UDAS = {
+        SUBJECT: {
+            'type': 'string',
+            'label': 'Debian BTS Subject',
+        },
+        URL: {
+            'type': 'string',
+            'label': 'Debian BTS URL',
+        },
+        NUMBER: {
+            'type': 'numeric',
+            'label': 'Debian BTS Number',
+        },
+        PACKAGE: {
+            'type': 'string',
+            'label': 'Debian BTS Package',
+        },
+        SOURCE: {
+            'type': 'string',
+            'label': 'Debian BTS Source Package',
+        },
+        FORWARDED: {
+            'type': 'string',
+            'label': 'Debian BTS Forwarded URL',
+        },
+    }
+    UNIQUE_KEY = (URL, )
+
+    PRIORITY_MAP = {
+        'wishlist': 'L',
+        'minor': 'L',
+        'normal': 'M',
+        'important': 'M',
+        'serious': 'H',
+        'grave': 'H',
+        'critical': 'H',
+    }
+
+    def to_taskwarrior(self):
+        return {
+            'priority': self.get_priority(),
+
+            self.URL: self.record['url'],
+            self.SUBJECT: self.record['subject'],
+            self.NUMBER: self.record['number'],
+            self.PACKAGE: self.record['package'],
+            self.SOURCE: self.record['source'],
+            self.FORWARDED: self.record['forwarded'],
+        }
+
+    def get_default_description(self):
+
+        return self.build_default_description(
+            title=self.record['subject'],
+            url=self.get_processed_url(self.record['url']),
+            number=self.record['number'],
+            cls='issue'
+        )
+
+    def get_priority(self):
+        return self.PRIORITY_MAP.get(
+            self.record.get('severity'),
+            self.origin['default_priority']
+        )
+
+
+class BTSService(IssueService, ServiceClient):
+    ISSUE_CLASS = BTSIssue
+    CONFIG_PREFIX = 'bts'
+
+    def __init__(self, *args, **kw):
+        super(BTSService, self).__init__(*args, **kw)
+        self.bts = debianbts
+        self.email = self.config_get_default('email', default=None)
+        self.packages = self.config_get_default('packages', default=None)
+        self.udd = self.config_get_default('udd', default=False, to_type=asbool)
+        self.udd_ignore_sponsor = self.config_get_default('udd_ignore_sponsor', default=True, to_type=asbool)
+        self.ignore_pkg = self.config_get_default('ignore_pkg', default=None)
+        self.ignore_src = self.config_get_default('ignore_src', default=None)
+
+    @classmethod
+    def validate_config(cls, config, target):
+#        if not config.has_option(target, 'trac.base_uri'):
+#            die("[%s] has no 'base_uri'" % target)
+#        elif '://' in config.get(target, 'trac.base_uri'):
+#            die("[%s] do not include scheme in 'base_uri'" % target)
+
+        IssueService.validate_config(config, target)
```

Comment:
Wow. Can't believe I missed that. I left that commented out to remind me to fill it in.

It essentially will just verify that bts.email is set to a valid email address, and that the true/false options can be parsed as such.

Setting back to [WIP].